lib/modules/platform/gitea/index.ts

```diff
@@ -173,6 +175,47 @@ async function lookupLabelByName(name: string): Promise<number | null> {
   return labelList.find((l) => l.name === name)?.id;
 }
 
+function getRepoUrl(
+  repo: helper.Repo,
+  gitUrl: GitUrlOption | undefined
+): string {
+  if (gitUrl === 'ssh') {
+    if (!repo.ssh_url) {
+      throw new Error(CONFIG_GIT_URL_UNAVAILABLE);
+    }
+    logger.debug({ url: repo.ssh_url }, `using ssh URL`);
+    return repo.ssh_url;
+  }
+
+  // Find options for current host and determine Git endpoint
+  const opts = hostRules.find({
+    hostType: PlatformId.Gitea,
+    url: defaults.endpoint,
+  });
+
+  if (gitUrl === 'endpoint' || repo.clone_url === null) {
+    if (repo.clone_url === null) {
+      logger.debug('no clone_url found. Falling back to old behaviour.');
+    }
+
+    const { protocol, host, pathname } = parseUrl(defaults.endpoint);
+    const newPathname = pathname.slice(0, pathname.indexOf('/api'));
+    const url = URL.format({
+      protocol: protocol.slice(0, -1) || 'https',
+      auth: 'token' + opts.token,
+      host,
+      pathname: newPathname + '/' + repo.full_name + '.git',
+    });
+    logger.debug({ url }, 'using URL based on configured endpoint');
+    return url;
+  }
+
+  logger.debug({ url: repo.clone_url }, `using http URL`);
```

Comment:
```suggestion
  logger.debug({ url: repo.clone_url }, `using HTTP URL`);
```

HTTP is an abbreviation, so should be capitalized. 😉